docs/user/interactive_environments.rst

```diff
@@ -0,0 +1,378 @@
+  installed in the environment, including the ``renku`` command-line installation.
+* ``.gitlab-ci.yml``: defines the docker build of the image based on the project's
+  ``Dockerfile``.
+
+
+
+Modifying the template's Dockerfile
+-----------------------------------
+
+The `Dockerfile` in the project is what defines the environment. In the template
+provided, the `Dockerfile`
+
+Renku Project uses Docker for containerization, which While we
+have a set of defaults that we build into a minimal Python and R images
+image, there are several reasons why you might want to build on top of these or
+write your own entirely.
+
+Dockerfile structure
+~~~~~~~~~~~~~~~~~~~~
+
+The project's ``Dockerfile`` lives in the top level of the project directory. In
+the default ``Dockerfile`` provided in the template, the first line is a ``FROM``
+statement that specifies a `versioned base docker image <https://github.com/SwissDataScienceCenter/renku-jupyter>`_.
+We add new versions periodically, but the heart of it is the set of installations
+of jupyterlab/rstudio, git, and renku::
+
+  FROM renku/singleuser:0.3.5-renku0.5.2
+
+  # or, for RStudio in the build
+
+  FROM renku/singleuser-r:0.3.5-renku0.5.2
+
+The next two statements install user-specified libraries from ``environment.yml``
+and ``requirements.txt``::
+
+  # install the python dependencies
+  COPY requirements.txt environment.yml /tmp/
+  RUN conda env update -q -f /tmp/environment.yml && \
+  /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+  conda clean -y --all && \
+  conda env export -n "root"
+
+You can add to this ``Dockerfile`` in any way you'd like.
+
+.. _docker_dev:
+
+Dockerfile development
+~~~~~~~~~~~~~~~~~~~~~~
+
+If you're going to be making simple modifications to the ``Dockerfile`` (i.e. changing
+the base Docker image version number), you can use the following steps to update
+and re-build the image:
+
+#. On the project's landing page, click the **View in GitLab** button in the upper righthand corner (opens a new tab by default).
+#. In GitLab, click the **Repository** tab in the lefthand column, which drops you into the **Files** tab.
+#. Click the **Dockerfile** out of the list of files that appears, and click **Edit** (top right, near the red Delete button. Don't click the red Delete button.)
+#. Make your edits in this window.
+#. When you're satisfied with the edits, scroll down and write a meaningful **commit message** (you'll thank yourself later).
+#. Click the green **Commit changes** button.
+
+Now you have committed the changes to your ``Dockerfile``. Since you have made a commit,
+the CI/CD pipeline will kick off (pre-configured for you as a ``renkulab-runner``
+inside the GitLab CI/CD settings). It will attempt to rebuild your project with
+the new contents of your ``Dockerfile`` based on the configuration in ``.gitlab-ci.yml``,
+a file at the top level of your project directory.
+
+The contents of ``.gitlab-ci.yml`` show you that in the build stage, we pull
```

Comment:
I think we should add here that normally people will not have to edit their `.gitlab-ci.yaml`.